ansible-scylla-node/tasks/Debian.yml

```diff
@@ -70,17 +70,76 @@
       state: latest
     when: scylla_version == 'latest' and scylla_edition == 'enterprise'
 
-  - name: Install specified OSS Scylla
-    apt:
-      name: "scylla={{ scylla_version }}*"
-      state: present
+  - name: Set Scylla {{ scylla_edition }} prefix
+    set_fact:
+      scylla_package_prefix: "scylla"
     when: scylla_version != 'latest' and scylla_edition == 'oss'
 
-  - name: Install specified Enterprise Scylla
-    apt:
-      name: "scylla-enterprise={{ scylla_version }}*"
-      state: present
+  - name: Set Scylla {{ scylla_edition }} prefix
+    set_fact:
+      scylla_package_prefix: "scylla-enterprise"
     when: scylla_version != 'latest' and scylla_edition == 'enterprise'
+
+  - name: Figure out exact Scylla version
+    block:
+    - name: Get versions of {{ scylla_package_prefix }} package
+      shell: apt list -a {{ scylla_package_prefix }} | grep {{ scylla_version }}
+      register: aptversions
+
+    - name: Search for version of {{ scylla_package_prefix }} package
+      set_fact:
+        tempsversion: "{{ aptversions.stdout | regex_findall(regexp) }}"
+      vars:
+          regexp: ' ([0-9\.]+)-([a-f0-9\.]+)-([0-9]) '
+
+    - name: Set {{ scylla_package_prefix }} version if only one version found
+      set_fact:
+        sversion: "{{ tempsversion }}"
+      when: tempsversion | length == 1
+
+    - name: Requery for {{ scylla_package_prefix }} version if multiple versions found
+      block:
+        - name: query for version with - at the end of {{ scylla_version }}
+          shell: apt list -a {{ scylla_package_prefix }} | grep {{ scylla_version }}-
```

Comment:
I guess you are trying to handle the case where you are either given the full scylla version like `2021.1.11-0.20220516.8e6c9917c` or a short version like `2021.1.11`, right?

I wonder why these two case can't be folded into a single regular expression?

Something like `egrep "{{ scylla_version }}[\-]?"`

![image](https://user-images.githubusercontent.com/6103981/175372391-57b75ac2-1521-43a4-b008-37bbb2382b10.png)

Reply:
reason was different, it was to error out if wrong version was provided (one that doesn't have a "-" at the end and doesn't match anything)
so basically it won't work for inputs like
2021
or 2021.1
and will error
so basically the "-" is optional in first test (since you can provide full version
2021.1.11-0.20220516.8e6c9917c-1 that will match in first test, but won't match in next one) , but a must match in second test, if first test won't work

so I don't know if above logic can be folded, if I'd fold it and use what you wrote it would continue even for 2021.1 and just pick more matches ... 
ok I guess the count of version test would then error out